xain/grpc/participant.py

```diff
@@ -65,14 +80,46 @@ def end_training(
     m = coordinator_pb2.EndTrainingRequest.Metrics(cid=cid, vol_by_class=vbc)
     # assemble req
     req = coordinator_pb2.EndTrainingRequest(
-        theta_update=theta_update_p, history=h, metrics=m
+        theta_update=theta_n_proto, history=h, metrics=m
     )
     # send request to end training
     reply = stub.EndTraining(req)
     print(f"Participant received: {type(reply)}")
 
 
+def init_participant() -> Participant:
+    xy_train_partitions, xy_val, _xy_test = load_splits(FLAGS.dataset)
+
+    model_fn = load_model_fn(FLAGS.model)
+    lr_fn_fn = load_lr_fn_fn(FLAGS.model)
+    model_provider = ModelProvider(model_fn, lr_fn_fn)
+
+    cid = 0
+    xy_train = xy_train_partitions[FLAGS.partition_id]
+    return Participant(
+        cid, model_provider, xy_train, xy_val, num_classes=10, batch_size=FLAGS.B
+    )
+
+
+def standby(channel, participant: Participant, terminate, selected):
+    # wait on heartbeat until *selected* event signals
+    while not selected.wait(RETRY_TIMEOUT):
+        print(f"Not yet selected for round. Retrying in {RETRY_TIMEOUT}s")
+    # ready:
+    theta, epochs, base = start_training(channel)
+    # training:
+    theta_n, his, _dict = participant.train_round(theta, epochs, base)
+    # NOTE _dict is the opt_config - ignore for now
+    met = participant.metrics()
+    end_training(channel, theta_n, his, met)
+    # back to standby unless terminate event says otherwise
+    if not terminate.isSet():
```

Comment:
We should prefer using `is_set` instead of `isSet`. That must be there for backwards compatibility https://github.com/python/cpython/blob/3.7/Lib/threading.py#L511